### ISSUE TITLE
Calls to settle must handle missing or old balance proofs 

### ISSUE THREAD

**Issue by hackaugusto** [raiden-network/raiden#2214]
## Problem Definition

> The state machine assumes the partner will always successfully call update transfer, and includes both balance proofs in the settlement call, if the update transaction fails then the settle transaction will also fail. To fix this the settle state change should not include the balance proof, but the balance hash, and the balance proof should be recovered from the WAL.

Additionally the state machine assumes always the latest balance proof is used, if an old balance proof is used the state changes in database needs to be searched to find the correct balance proof data

related to: #2129 

Split it into two issues:

- https://github.com/raiden-network/raiden/issues/2410
- https://github.com/raiden-network/raiden/issues/2414

**Comment by palango** [comment]
This is blocked by the conversion to json state which is part of #862

**Comment by LefterisJP** [comment]
Split this into two issues and updated the description. When they are done this is done. We can perhaps close this even.

**Comment by hackaugusto** [comment]
the two sub issues have been implemented.
